src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -885,11 +881,25 @@ extern "C" UInt32_BOOL ResetEvent(HANDLE event)
     return UInt32_TRUE;
 }
 
-extern "C" UInt32 GetEnvironmentVariableA(const char * pName, char * pBuffer, UInt32 size)
+extern "C" UInt32 GetEnvironmentVariableA(const char * name, char * buffer, UInt32 size)
 {
-    // UNIXTODO: Implement this function
-    *pBuffer = '\0';
-    return 0;
+    // Using std::getenv instead of getenv since it is guaranteed to be thread safe w.r.t. other
+    // std::getenv calls in C++11
+    const char* value = std::getenv(name);
+    if (value == NULL)
+    {
+        return 0;
+    }
+
+    size_t valueLen = strlen(value);
+    if (valueLen < size)
+    {
+        strcpy(buffer, value);
+        return valueLen;
+    }
+
+    // return required size including the null character
+    return valueLen + 1;
```

Comment:
size_t -> UInt32 conversion can overflow. (It is likely a theoretical chance here ... but the C++ sanitizers tend to complain about it regardless.)